diff -Naur a/5.15/drivers/iio/accel/hid-sensor-accel-3d.c b/5.15/drivers/iio/accel/hid-sensor-accel-3d.c
--- a/5.15/drivers/iio/accel/hid-sensor-accel-3d.c	2021-10-31 20:53:10.000000000 +0000
+++ b/5.15/drivers/iio/accel/hid-sensor-accel-3d.c	2022-08-04 17:12:42.387992772 +0100
@@ -460,4 +460,5 @@
 MODULE_DESCRIPTION("HID Sensor Accel 3D");
 MODULE_AUTHOR("Srinivas Pandruvada <srinivas.pandruvada@intel.com>");
 MODULE_LICENSE("GPL");
 MODULE_IMPORT_NS(IIO_HID);
+MODULE_VERSION("slamcore");
diff -Naur a/5.15/drivers/iio/gyro/hid-sensor-gyro-3d.c b/5.15/drivers/iio/gyro/hid-sensor-gyro-3d.c
--- a/5.15/drivers/iio/gyro/hid-sensor-gyro-3d.c	2021-10-31 20:53:10.000000000 +0000
+++ b/5.15/drivers/iio/gyro/hid-sensor-gyro-3d.c	2022-08-04 17:13:24.628948294 +0100
@@ -394,4 +394,5 @@
 MODULE_DESCRIPTION("HID Sensor Gyroscope 3D");
 MODULE_AUTHOR("Srinivas Pandruvada <srinivas.pandruvada@intel.com>");
 MODULE_LICENSE("GPL");
 MODULE_IMPORT_NS(IIO_HID);
+MODULE_VERSION("slamcore");
diff -Naur a/5.15/drivers/media/usb/uvc/Makefile b/5.15/drivers/media/usb/uvc/Makefile
--- a/5.15/drivers/media/usb/uvc/Makefile	2021-10-31 20:53:10.000000000 +0000
+++ b/5.15/drivers/media/usb/uvc/Makefile	2022-08-04 17:11:42.474639280 +0100
@@ -5,3 +5,8 @@
 uvcvideo-objs  += uvc_entity.o
 endif
 obj-$(CONFIG_USB_VIDEO_CLASS) += uvcvideo.o
+
+#Librealsense-dkms addition
+CONFIG_MODULE_SIG=n
+CC := ${CC} -I ${M}/../../../../include-overrides
+ccflags-y := -DDRIVER_VERSION_SUFFIX=\"${DRIVER_VERSION_SUFFIX}\"
diff -Naur a/5.15/drivers/media/usb/uvc/uvcvideo.h b/5.15/drivers/media/usb/uvc/uvcvideo.h
--- a/5.15/drivers/media/usb/uvc/uvcvideo.h	2022-08-04 16:57:18.000000000 +0100
+++ b/5.15/drivers/media/usb/uvc/uvcvideo.h	2022-08-04 17:11:42.474639280 +0100
@@ -180,8 +180,13 @@
  * Driver specific constants.
  */
 
-#define DRIVER_VERSION		"1.1.1"
+/* Librealsense dkms amendments*/
+#ifndef DRIVER_VERSION_SUFFIX
+#define DRIVER_VERSION_SUFFIX
+#endif
 
+#define DRIVER_VERSION		"1.1.2" DRIVER_VERSION_SUFFIX
+ 
 /* Number of isochronous URBs. */
 #define UVC_URBS		5
 /* Maximum number of packets per URB. */
diff -Naur a/5.15/drivers/media/v4l2-core/Makefile b/5.15/drivers/media/v4l2-core/Makefile
--- a/5.15/drivers/media/v4l2-core/Makefile	2021-10-31 20:53:10.000000000 +0000
+++ b/5.15/drivers/media/v4l2-core/Makefile	2022-08-04 17:11:42.474639280 +0100
@@ -2,6 +2,8 @@
 #
 # Makefile for the V4L2 core
 #
+# Librealsense dkms appendix
+CC := ${CC} -I ${M}/../../../include-overrides -I ${M}/../tuners -I ${M}/../dvb-core -I ${M}/../dvb-frontends
 
 tuner-objs	:=	tuner-core.o
 
diff -Naur a/5.15/drivers/media/v4l2-core/v4l2-dev.c b/5.15/drivers/media/v4l2-core/v4l2-dev.c
--- a/5.15/drivers/media/v4l2-core/v4l2-dev.c	2021-10-31 20:53:10.000000000 +0000
+++ b/5.15/drivers/media/v4l2-core/v4l2-dev.c	2022-08-04 17:11:42.474639280 +0100
@@ -1136,3 +1136,4 @@
 MODULE_DESCRIPTION("Video4Linux2 core driver");
 MODULE_LICENSE("GPL");
 MODULE_ALIAS_CHARDEV_MAJOR(VIDEO_MAJOR);
+MODULE_VERSION("slamcore");
