diff -Naur a/5.15/drivers/iio/accel/hid-sensor-accel-3d.c b/5.15/drivers/iio/accel/hid-sensor-accel-3d.c
--- a/5.15/drivers/iio/accel/hid-sensor-accel-3d.c	2021-10-31 20:53:10.000000000 +0000
+++ b/5.15/drivers/iio/accel/hid-sensor-accel-3d.c	2022-08-04 17:04:00.424208999 +0100
@@ -280,6 +280,7 @@
 			hid_sensor_convert_timestamp(
 					&accel_state->common_attributes,
 					*(int64_t *)raw_data);
+		ret = 0;
 	break;
 	default:
 		break;
diff -Naur a/5.15/drivers/iio/gyro/hid-sensor-gyro-3d.c b/5.15/drivers/iio/gyro/hid-sensor-gyro-3d.c
--- a/5.15/drivers/iio/gyro/hid-sensor-gyro-3d.c	2021-10-31 20:53:10.000000000 +0000
+++ b/5.15/drivers/iio/gyro/hid-sensor-gyro-3d.c	2022-08-04 17:04:00.424208999 +0100
@@ -231,6 +231,7 @@
 		gyro_state->timestamp =
 			hid_sensor_convert_timestamp(&gyro_state->common_attributes,
 						     *(s64 *)raw_data);
+		ret = 0;
 	break;
 	default:
 		break;
