diff -Naur a/linux/drivers/iio/accel/hid-sensor-accel-3d.c b/linux/drivers/iio/accel/hid-sensor-accel-3d.c
--- a/linux/drivers/iio/accel/hid-sensor-accel-3d.c	2022-04-11 14:06:52.825901912 +0000
+++ b/linux/drivers/iio/accel/hid-sensor-accel-3d.c	2022-04-11 14:09:52.166105591 +0000
@@ -283,6 +283,7 @@
 			hid_sensor_convert_timestamp(
 					&accel_state->common_attributes,
 					*(int64_t *)raw_data);
+		ret = 0;
 	break;
 	default:
 		break;
diff -Naur a/linux/drivers/iio/gyro/hid-sensor-gyro-3d.c b/linux/drivers/iio/gyro/hid-sensor-gyro-3d.c
--- a/linux/drivers/iio/gyro/hid-sensor-gyro-3d.c	2022-04-11 14:06:52.833902099 +0000
+++ b/linux/drivers/iio/gyro/hid-sensor-gyro-3d.c	2022-04-11 14:10:21.410791085 +0000
@@ -234,6 +234,7 @@
 		gyro_state->timestamp =
 			hid_sensor_convert_timestamp(&gyro_state->common_attributes,
 						     *(s64 *)raw_data);
+		ret = 0;
 	break;
 	default:
 		break;
