diff -Naur a/tegra/drivers/hid/Makefile b/tegra/drivers/hid/Makefile
--- a/tegra/drivers/hid/Makefile	2023-12-20 14:12:45.576120607 +0000
+++ b/tegra/drivers/hid/Makefile	2023-12-20 15:31:31.785814130 +0000
@@ -134,6 +134,9 @@
 obj-$(CONFIG_HID_WACOM)		+= wacom.o
 obj-$(CONFIG_HID_WALTOP)	+= hid-waltop.o
 obj-$(CONFIG_HID_WIIMOTE)	+= hid-wiimote.o
+
+CONFIG_HID_SENSOR_HUB=m
+
 obj-$(CONFIG_HID_SENSOR_HUB)	+= hid-sensor-hub.o
 obj-$(CONFIG_HID_SENSOR_CUSTOM_SENSOR)	+= hid-sensor-custom.o
 
diff -Naur a/tegra/drivers/iio/accel/hid-sensor-accel-3d.c b/tegra/drivers/iio/accel/hid-sensor-accel-3d.c
--- a/tegra/drivers/iio/accel/hid-sensor-accel-3d.c	2023-12-20 14:12:45.580120697 +0000
+++ b/tegra/drivers/iio/accel/hid-sensor-accel-3d.c	2023-12-20 15:30:43.044721494 +0000
@@ -462,3 +462,4 @@
 MODULE_AUTHOR("Srinivas Pandruvada <srinivas.pandruvada@intel.com>");
 MODULE_LICENSE("GPL");
 MODULE_IMPORT_NS(IIO_HID);
+MODULE_VERSION("slamcore");
diff -Naur a/tegra/drivers/iio/accel/Makefile b/tegra/drivers/iio/accel/Makefile
--- a/tegra/drivers/iio/accel/Makefile	2023-12-20 14:12:45.580120697 +0000
+++ b/tegra/drivers/iio/accel/Makefile	2023-12-20 15:33:01.743830725 +0000
@@ -30,6 +30,9 @@
 obj-$(CONFIG_FXLS8962AF)	+= fxls8962af-core.o
 obj-$(CONFIG_FXLS8962AF_I2C)	+= fxls8962af-i2c.o
 obj-$(CONFIG_FXLS8962AF_SPI)	+= fxls8962af-spi.o
+
+CONFIG_HID_SENSOR_ACCEL_3D=m
+
 obj-$(CONFIG_HID_SENSOR_ACCEL_3D) += hid-sensor-accel-3d.o
 obj-$(CONFIG_KXCJK1013) += kxcjk-1013.o
 obj-$(CONFIG_KXSD9)	+= kxsd9.o
diff -Naur a/tegra/drivers/iio/common/hid-sensors/Makefile b/tegra/drivers/iio/common/hid-sensors/Makefile
--- a/tegra/drivers/iio/common/hid-sensors/Makefile	2023-12-20 14:12:45.584120785 +0000
+++ b/tegra/drivers/iio/common/hid-sensors/Makefile	2023-12-20 18:00:19.123612624 +0000
@@ -3,6 +3,9 @@
 # Makefile for the Hid sensor common modules.
 #
 
+CONFIG_HID_SENSOR_IIO_COMMON=m
+CONFIG_HID_SENSOR_IIO_TRIGGER=m
+
 obj-$(CONFIG_HID_SENSOR_IIO_COMMON) += hid-sensor-iio-common.o
 obj-$(CONFIG_HID_SENSOR_IIO_TRIGGER) += hid-sensor-trigger.o
 hid-sensor-iio-common-y := hid-sensor-attributes.o
diff -Naur a/tegra/drivers/iio/gyro/hid-sensor-gyro-3d.c b/tegra/drivers/iio/gyro/hid-sensor-gyro-3d.c
--- a/tegra/drivers/iio/gyro/hid-sensor-gyro-3d.c	2023-12-20 14:12:45.584120785 +0000
+++ b/tegra/drivers/iio/gyro/hid-sensor-gyro-3d.c	2023-12-20 15:30:43.044721494 +0000
@@ -396,3 +396,4 @@
 MODULE_AUTHOR("Srinivas Pandruvada <srinivas.pandruvada@intel.com>");
 MODULE_LICENSE("GPL");
 MODULE_IMPORT_NS(IIO_HID);
+MODULE_VERSION("slamcore");
diff -Naur a/tegra/drivers/iio/gyro/Makefile b/tegra/drivers/iio/gyro/Makefile
--- a/tegra/drivers/iio/gyro/Makefile	2023-12-20 14:12:45.584120785 +0000
+++ b/tegra/drivers/iio/gyro/Makefile	2023-12-20 15:33:45.728816739 +0000
@@ -17,6 +17,8 @@
 obj-$(CONFIG_FXAS21002C_I2C) += fxas21002c_i2c.o
 obj-$(CONFIG_FXAS21002C_SPI) += fxas21002c_spi.o
 
+CONFIG_HID_SENSOR_GYRO_3D=m
+
 obj-$(CONFIG_HID_SENSOR_GYRO_3D) += hid-sensor-gyro-3d.o
 
 # Currently this is rolled into one module, split it if
diff -Naur a/tegra/drivers/media/usb/uvc/Makefile b/tegra/drivers/media/usb/uvc/Makefile
--- a/tegra/drivers/media/usb/uvc/Makefile	2023-12-20 14:12:45.600121142 +0000
+++ b/tegra/drivers/media/usb/uvc/Makefile	2023-12-20 15:30:43.044721494 +0000
@@ -5,3 +5,8 @@
 uvcvideo-objs  += uvc_entity.o
 endif
 obj-$(CONFIG_USB_VIDEO_CLASS) += uvcvideo.o
+
+#Librealsense-dkms addition
+CONFIG_MODULE_SIG=n
+CC := ${CC} -I ${M}/../../../../include-overrides
+ccflags-y := -DDRIVER_VERSION_SUFFIX=\"${DRIVER_VERSION_SUFFIX}\"
diff -Naur a/tegra/drivers/media/usb/uvc/uvcvideo.h b/tegra/drivers/media/usb/uvc/uvcvideo.h
--- a/tegra/drivers/media/usb/uvc/uvcvideo.h	2023-12-20 14:12:45.600121142 +0000
+++ b/tegra/drivers/media/usb/uvc/uvcvideo.h	2023-12-20 15:30:43.044721494 +0000
@@ -180,8 +180,13 @@
  * Driver specific constants.
  */
 
-#define DRIVER_VERSION		"1.1.1"
+/* Librealsense dkms amendments*/
+#ifndef DRIVER_VERSION_SUFFIX
+#define DRIVER_VERSION_SUFFIX
+#endif
 
+#define DRIVER_VERSION		"1.1.2" DRIVER_VERSION_SUFFIX
+ 
 /* Number of isochronous URBs. */
 #define UVC_URBS		5
 /* Maximum number of packets per URB. */
diff -Naur a/tegra/drivers/media/v4l2-core/Makefile b/tegra/drivers/media/v4l2-core/Makefile
--- a/tegra/drivers/media/v4l2-core/Makefile	2023-12-20 14:12:45.600121142 +0000
+++ b/tegra/drivers/media/v4l2-core/Makefile	2023-12-20 15:30:43.044721494 +0000
@@ -2,6 +2,8 @@
 #
 # Makefile for the V4L2 core
 #
+# Librealsense dkms appendix
+CC := ${CC} -I ${M}/../../../include-overrides -I ${M}/../tuners -I ${M}/../dvb-core -I ${M}/../dvb-frontends
 
 tuner-objs	:=	tuner-core.o
 
@@ -9,6 +11,9 @@
 			v4l2-event.o v4l2-subdev.o v4l2-common.o \
 			v4l2-ctrls-core.o v4l2-ctrls-api.o \
 			v4l2-ctrls-request.o v4l2-ctrls-defs.o
+
+CONFIG_VIDEO_V4L2=m
+
 videodev-$(CONFIG_COMPAT) += v4l2-compat-ioctl32.o
 videodev-$(CONFIG_TRACEPOINTS) += v4l2-trace.o
 videodev-$(CONFIG_MEDIA_CONTROLLER) += v4l2-mc.o
diff -Naur a/tegra/drivers/media/v4l2-core/v4l2-dev.c b/tegra/drivers/media/v4l2-core/v4l2-dev.c
--- a/tegra/drivers/media/v4l2-core/v4l2-dev.c	2023-12-20 14:12:45.600121142 +0000
+++ b/tegra/drivers/media/v4l2-core/v4l2-dev.c	2023-12-20 15:30:43.044721494 +0000
@@ -1136,3 +1136,4 @@
 MODULE_DESCRIPTION("Video4Linux2 core driver");
 MODULE_LICENSE("GPL");
 MODULE_ALIAS_CHARDEV_MAJOR(VIDEO_MAJOR);
+MODULE_VERSION("slamcore");
